saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/web/authentication/OpenSamlAuthenticationRequestResolver.java

```diff
@@ -122,7 +129,7 @@
 		if (authnRequest.getID() == null) {
 			authnRequest.setID("ARQ" + UUID.randomUUID().toString().substring(1));
 		}
-		String relayState = UUID.randomUUID().toString();
+		String relayState = relayStateResolver.convert(request);
```

Comment:
Please change to `this.relayStateResolver...` to fit with Spring Security's coding conventions.